variables.tf

```diff
@@ -363,6 +363,35 @@ variable "security_group_rules" {
   }
 }
 
+variable "clean_default_security_group" {
+  description = "Remove all rules from the default VPC security group (less permissive)"
+  type        = bool
+  default     = false
+}
+
+variable "clean_default_acl" {
+  description = "Remove all rules from the default VPC ACL (less permissive)"
+  type        = bool
+  default     = false
+}
+
+variable "ibmcloud_api_visibility" {
+  description = "IBM Cloud API visibility used by scripts run in this module. Must be 'public', 'private', or 'public-and-private'"
+  type        = string
+  default     = "public"
+
+  validation {
+    error_message = "IBM Cloud API visibility must be either 'public', 'private', or 'public-and-private'"
+    condition     = (var.ibmcloud_api_visibility == "public") || (var.ibmcloud_api_visibility == "private") || (var.ibmcloud_api_visibility == "public-and-private")
```

Comment:
what does `public-and-private` do? Should it even be an option here?

Reply:
I have set this variable to exactly match the `visibility` parameter of the IBM provider, and those are the three values. My thought was that if someone has a variable in their scaffold already to set that attribute, they can send same value to this module to do the same thing. My script doesn't differentiate between "private" and "public-and-private" because I know that the few things I'm using all support private. I think the provider uses "public-and-private" to use global endpoints if one of the services doesn't support private, but "private" would just error out.
Reference: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs#visibility